docs/source/examples/managed-jobs.rst

```diff
@@ -282,6 +282,24 @@ candidate resources for a job. See documentation :ref:`here
 In this example, SkyPilot will perform cost optimizations to select the resource to use, which almost certainly
 will be spot instances. If spot instances are not available, SkyPilot will fall back to launch on-demand instances.
 
+
+Jobs Restarts on User Code Failure
+-----------------------------------
+
+By default, SkyPilot will try to recover a job when its underlying cluster is preempted or failed. Any user code failures (non-zero exit codes) are not auto-recovered.
+
+In some cases, you may want a job to automatically restart on its own failures, e.g., when a training job crashes due to a Nvidia driver issue or NCCL timeouts. To specify this, you
+can further set :code:`max_restarts_on_failure` in :code:`resources.job_recovery` in the job YAML file.
+
+.. code-block:: yaml
+
+  resources:
+    accelerators: A100:8
+    job_recovery:
+      # Restart the job up to 3 times on user code failure.
+      max_restarts_on_failure: 3
```

Comment:
I'm slightly confused by the naming. Just looking at `max_restarts_on_failure` makes me think it also applies to spot instance failure (does it apply? or are spot instance failure max restarts infinite?).

Perhaps a clearer way would be to have two distinct fields like so:
```
  resources:
    accelerators: A100:8
    job_recovery:
      restart_on_job_errors: bool
      max_restarts_on_failure: int, defaults to infinite
```

wdyt? cc @concretevitamin

Reply:
How about `max_restarts_on_user_failure`? I found having two fields a bit verbose. Wdyt?

Reply:
Hmm I think two fields is clearer and clarifies the intent. Not feeling too strongly though. 

If we do decide on having a single field, how about `max_restarts_on_job_errors`?

Reply:
Discussed offline, we will use `max_restarts_on_errors`